### PR TITLE
va-textarea: add support for formHeadingLevel prop

### PR DESCRIPTION
**Pull request by powellkerry**
## Chromatic

https://2974-textarea-header-level--65a6e2ed2314f7b8f98609d8.chromatic.com

## Description
Add support to change the header level in va-textarea

Closes [2974](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2974)

## QA Checklist
- [ x] Component maintains 1:1 parity with design mocks
- [ x] Text is consistent with what's been provided in the mocks
- [ x] Component behaves as expected across breakpoints
- [x] Accessibility expert has signed off on code changes (if applicable. If not applicable provide reason why)
- [x] Designer has signed off on changes (if applicable. If not applicable provide reason why)
- [ x] Tab order and focus state work as expected
- [ x] Changes have been tested against screen readers (if applicable. If not applicable provide reason why)
- [ x] New components are covered by e2e tests; updates to existing components are covered by existing test suite
- [ x] Changes have been tested in vets-website using Verdaccio (if applicable. If not applicable provide reason why)

## Screenshots
Screenshot 2024-07-18 at 3 23 52 PM

## Acceptance criteria
- [x] QA checklist has been completed
- [x] Screenshots have been attached that cover desktop and mobile screens

## Definition of done
- [ x] Documentation has been updated, if applicable
- [ x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

File: packages/web-components/src/components/va-textarea/va-textarea.tsx [hunk 1]
```diff
@@ -168,6 +179,11 @@ export class VaTextarea {
     return this.maxlength;
   }
 
+  private getHeaderLevel() {
```
Comment (jamigibbs) [inline packages/web-components/src/components/va-textarea/va-textarea.tsx:182]: Can this be imported from the utilities?

`import { getHeaderLevel } from '../../utils/utils';`

File: packages/web-components/src/components/va-textarea/va-textarea.tsx [hunk 2]
```diff
@@ -233,14 +248,25 @@ export class VaTextarea {
...
+                    </span>
+                  )
+                }
+                {headerAriaDescribedby && (
+                  <span id="header-message" class="sr-only">
+                    {headerAriaDescribedby}
                   </span>
                 )}
-                {hint && <span class="usa-hint">{hint}</span>}
+                {required && <span class="usa-label--required" part="required">{i18next.t('required')}</span>}
```
Comment (jamigibbs) [inline packages/web-components/src/components/va-textarea/va-textarea.tsx:268]: Was this `part="required"` added intentionally?

**Review by jamigibbs — APPROVED** [review]
Thanks for doing this work, Kerry! I just have a question about adding `part` to the required text.

File: packages/web-components/src/components/va-textarea/va-textarea.tsx (hunk 2 above)
Comment (powellkerry) [reply packages/web-components/src/components/va-textarea/va-textarea.tsx:268]: No it was copied. I'll remove it.

**Review by harshil1793 — APPROVED** [review]

**Review by rsmithadhoc — APPROVED** [review]

**Pull request merged**